Dockerfile

```diff
@@ -31,9 +42,15 @@ COPY --from=builder /opt/restic /opt/restic
 RUN chmod +x /opt/restic && \
     ln -s /opt/restic /usr/bin
 
-ENTRYPOINT ["/opt/backup-loop.sh"]
+COPY --from=builder /opt/entrypoint-demoter/entrypoint-demoter /opt/entrypoint-demoter
 
-VOLUME ["/data", "/backups"]
+RUN chmod +x /opt/entrypoint-demoter && \
+    ln -s /opt/entrypoint-demoter /usr/bin
 
 COPY backup-loop.sh /opt/
 
+VOLUME ["/data", "/backups"]
+
+ENTRYPOINT ["/opt/entrypoint-demoter", "--match", "/data"]
```

Comment:
I'm thinking this should match `/backups` since that's the one that needs to be writable by the container's user, if that option is used.

Reply:
Remember that `/backups` might not always exist / be mounted (i.e. using external bucket, using different path etc). This is why I think this should be more dynamically decided what arguments to run.
This is an example, where /backups isn't mounted:
```
bash-5.0# ps aux
PID   USER     TIME  COMMAND
    1 root      0:00 /opt/entrypoint-demoter --match /backups /opt/backup-loop.sh
   10 root      0:00 {backup-loop.sh} /bin/bash /opt/backup-loop.sh
  157 root      0:00 sleep 1h
```

Reply:
True, `/data` would always be attached since that's the absolute requirement of the backup operation.

Reply:
Actually yes and no. Since in the Dockerfile `/data` and `/backups` are mentioned, it'll always create custom volumes for them, if no volumes override them. Example:
```
docker inspect deploy-backup | jq '.[0].Mounts | [ .[] | {Type, Name, Destination} ]'
[
  {
    "Type": "volume",
    "Name": "e719c8cf78817b70961c83760dd3f7146fc1fc0e7e7acbe3fc73fcd701b2b7fe",
    "Destination": "/data"
  },
  {
    "Type": "volume",
    "Name": "f1169471a1175a27ecb475aebae84b9e9d2d0ffbfd3b8af945c2ae0971f85b6d",
    "Destination": "/backups"
  },
  {
    "Type": "volume",
    "Name": "deploy-data",
    "Destination": "/something/data"
  }
]
```
This means that these directories will be mounted as root iirc.
Also, remember that one can use `SRC_DIR` to change the location, so it might be something else. I'd be for removing volumes from Dockerfile (or leaving at most `/backups`) and creating tinny entrypoint wrapper around the demoter to dynamically decide what arguments to pass (i.e. use environment variables passed to it - but then we'd need to duplicate location of some environment variable defaults :/).